src/Interval.js

```diff
@@ -86,6 +87,19 @@ Interval.prototype = {
     }
   },
 
+  // Returns a new Interval that has the same extent as this one, but which is relative
+  // to `that`, an Interval that fully covers this one.
+  relativeTo: function(that, newInputStream) {
+    if (this.inputStream !== that.inputStream) {
+      throw errors.intervalSourcesDontMatch();
+    }
+    assert(this.startIdx >= that.startIdx && this.endIdx <= that.endIdx,
```

Comment:
It's the latter, basically. It should only if a recipe contains a rule whose interval (as specified by `.at(x, y)` is outside the bounds of the source that is included in that recipe. The only way that would happen is either (a) a bug, or (b) someone edited the recipe by hand (which they shouldn't do).